src/@commonsku/styles/Thermometer.tsx

```diff
@@ -1,122 +1,152 @@
+    padding-left: ${props => `${props.containerWidth - props.targetWidth}px`};
+`;
 
 export type ThermometerProps = {
     style?: React.CSSProperties;
     title?: string;
     target: number;
-    value1: number,
-    value1Label?: string | ((v: number) => string),
-	barColor?: string;
-	labelTextColor?: string;
-	isSecondary?: boolean;
+    value: number,
+    valueLabel?: React.ReactNode | ((v: number) => React.ReactNode),
+    targetLabel?: React.ReactNode | ((v: number) => React.ReactNode),
+    barColor?: string;
+    labelTextColor?: string;
+    isSecondary?: boolean;
 };
+
 export default function Thermometer({
     title,
     target,
-    value1,
-    value1Label,
-	barColor,
-	labelTextColor,
-	isSecondary,
+    targetLabel,
+    value,
+    valueLabel,
+    barColor,
+    labelTextColor,
+    isSecondary,
     ...props
 }: ThermometerProps) {
-    const targetRef = useRef<HTMLSpanElement | null>(null);
-    const val1Ref = useRef<HTMLSpanElement | null>(null);
     const containerRef = useRef<HTMLDivElement | null>(null);
+    const targetRef = useRef<HTMLSpanElement | null>(null);
     const [containerWidth, setContainerWidth] = useState(0);
     const [targetWidth, setTargetWidth] = useState(0);
-    const [value1Width, setValue1Width] = useState(0);
 
-    const calcTargetWidth = () => {
-        const result = 1 * containerWidth;
-        return result > containerWidth ? containerWidth : result;
-    };
-    const calcVal1Width = () => {
-        const result = (target <= value1 ? 1 : value1/target) * containerWidth;
-        return result > containerWidth ? containerWidth : result;
-    };
+    useLayoutEffect(() => {
+        setContainerWidth(containerRef.current?.offsetWidth ?? 0);
+        setTargetWidth(targetRef.current?.offsetWidth ?? 0);
 
-    const measureContainerRef = useCallback((node: HTMLDivElement) => {
-        containerRef.current = node;
-        setContainerWidth(node?.clientWidth || 0);
-    }, []);
-    const measureTargetRef = useCallback((node: HTMLSpanElement) => {
-        targetRef.current = node;
-        setTargetWidth(node?.clientWidth || 0);
-    }, []);
-    const measureValue1Ref = useCallback((node: HTMLSpanElement) => {
-        val1Ref.current = node;
-        setValue1Width(node?.clientWidth || 0);
+        const onResize = () => {
+            if (containerRef.current === null || targetRef.current === null) {
+                return;
+            }
+
+            setContainerWidth(containerRef.current.offsetWidth);
+
+            const targetPadding = getComputedStyle(targetRef.current).paddingLeft;
+            const targetWidth = targetRef.current.offsetWidth - parseInt(targetPadding);
+
+            setTargetWidth(targetWidth);
+        };
+        window.addEventListener("resize", onResize);
+
+        return () => window.removeEventListener("resize", onResize);
     }, []);
 
+    const getProgressWidth = () => {
+        const progress = Math.min(1, value / target);
```

Comment:
Can you add something to handle the case where `target === 0` i.e `Number.isNaN(value / target) === true`